src/lectern-client/schema-functions.ts

```diff
@@ -57,34 +57,72 @@ export const process = (
     throw new Error(`no schema found for : ${definition}`);
   }
 
+  let validationErrors: SchemaValidationError[] = [];
+  let processedRecords: TypedDataRecord[] = [];
+
+  records.forEach((r, i) => {
+    const result = process(dataSchema, definition, r, i);
+    validationErrors = validationErrors.concat(result.validationErrors);
+    processedRecords = processedRecords.concat(result.processedRecords);
+  });
+
+  L.debug(
+    `done processing all rows, validationErrors: ${validationErrors.length}, validRecords: ${processedRecords.length}`,
+  );
+
+  return F({
+    validationErrors,
+    processedRecords,
+  });
+};
+
+export const process = (
```

Comment:
a new method to process one record instead of batch but keeps index information

Reply:
cool, but it looks a little weird, this function is processing one record, but it returns `prcessedRecords` which is an array that only contains the one record?? guess it makes it easy if both functions return arrays... Is that what you were going for?

Reply:
I'll take a look .. I don't think I left it like that for a reason